hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/marker/WriteMarkers.java

```diff
@@ -107,6 +107,34 @@ public Option<Path> createIfNotExists(String partitionPath, String fileName, IOT
     return create(partitionPath, fileName, type, true);
   }
 
+  /**
+   * Creates a marker if the marker does not exist.
+   * This can invoke marker-based early conflict detection when enabled for multi-writers.
+   *
+   * @param partitionPath  partition path in the table
+   * @param fileName       file name
+   * @param type           write IO type
+   * @param writeConfig    Hudi write configs.
+   * @param fileId         File ID.
+   * @param activeTimeline Active timeline for the write operation.
+   * @return the marker path.
+   */
+  public Option<Path> createIfNotExists(String partitionPath, String fileName, IOType type, HoodieWriteConfig writeConfig,
+                             String fileId, HoodieActiveTimeline activeTimeline) {
+    if (writeConfig.getWriteConcurrencyMode().supportsOptimisticConcurrencyControl()
+        && writeConfig.isEarlyConflictDetectionEnable()) {
```

Comment:
can we change it to

```writeConfig.isEarlyConflictDetectionEnable() 
     && writeConfig.getWriteConcurrencyMode().supportsOptimisticConcurrencyControl()
```

which is more efficient.

Reply:
Swap them in new commit.